vmcnet/train/runners.py

```diff
@@ -635,29 +631,41 @@ def run_molecule() -> None:
         apply_pmap=config.distribute,
     )
 
+    start_epoch = 0
+
     if reload_from_checkpoint:
         checkpoint_file_path = os.path.join(
             reload_config.logdir, reload_config.checkpoint_relative_file_path
         )
         directory, filename = os.path.split(checkpoint_file_path)
 
-        if reload_run_from_checkpoint:
-            _, data, params, optimizer_state, key = utils.io.reload_vmc_state(
-                directory, filename
-            )
-            (
-                data,
-                params,
-                optimizer_state,
-                key,
-            ) = utils.distribute.distribute_vmc_state_from_checkpoint(
-                data, params, optimizer_state, key
+        (
+            reload_at_epoch,
+            data,
+            params,
+            reloaded_optimizer_state,
+            key,
+        ) = utils.io.reload_vmc_state(directory, filename)
+
+        if reload_config.append:
+            start_epoch = reload_at_epoch
```

Comment:
One thing to note is that this change of `start_epoch` is actually a very important effect of the `append` flag, since this affects not just the logging but also the learning rate that is used. In other words, whether or not you want the text files to be concatenated, the only way to get the behavior of truly continuing a previous run, with the same learning rate schedule it would have had, is to use `reload_config.append`. I think that's probably fine but just wanted to point it out.

Reply:
Idk maybe we should separate out those two pieces into different flags or rename this flag to something like `reload_config.continue_epoch_count.` Not totally sure.

Reply:
I've changed it to use the advanced start_epoch iff the old optimizer state is used